code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_miner.dm

```diff
@@ -160,11 +160,9 @@ Difficulty: Medium
 		changeNext_move(CLICK_CD_RANGE)
 
 /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/proc/quick_attack_loop()
-	if(next_move <= world.time)
-		stoplag(1)
-		.() //retry
-		return
-	sleep((next_move - world.time) * 1.5)
+	while(next_move <= world.time) //this is done this way because next_move can change to be sooner while we sleep.
+		stoplag(1) 
```

Comment:
this should probably exit out if you have no target, since it didn't do that before